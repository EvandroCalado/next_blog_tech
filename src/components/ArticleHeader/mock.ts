export default {
  id: '3',
  title: 'Desenvolvimento de aplicativos web escaláveis com Next.js',
  excerpt:
    'Se você está procurando um framework poderoso para desenvolver aplicativos web escaláveis, o Next.js pode ser a escolha certa para você. O Next.js é um framework baseado em React que oferece recursos avançados, como renderização no lado do servidor e roteamento dinâmico, para ajudá-lo a criar aplicativos web modernos e de alto desempenho. Neste post, exploraremos as principais características do Next.js e veremos como usá-lo para construir aplicativos web escaláveis.',
  createdAt: '2023-03-27T17:22:39.749Z',
  cover:
    'https://res.cloudinary.com/dnhsdenji/image/upload/v1679929576/twitter_card_f0df8b6a20.png',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
