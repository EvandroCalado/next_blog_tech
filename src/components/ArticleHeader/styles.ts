import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.small} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};

    @media ${theme.media.small} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
