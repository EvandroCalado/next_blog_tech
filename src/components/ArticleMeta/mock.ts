export default {
  createdAt: '2023-03-27T17:22:39.749Z',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
