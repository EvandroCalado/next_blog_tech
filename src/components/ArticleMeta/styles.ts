import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child)::after {
      content: ', ';
    }

    span:first-child {
      margin-right: 0.5rem;
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
