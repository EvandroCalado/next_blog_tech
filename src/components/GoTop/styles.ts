import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    color: ${theme.colors.primary};
    border: 1px solid ${theme.colors.primary};
    border-radius: 50%;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 5rem;
    height: 5rem;
    z-index: 6;
    text-decoration: none;
  `}
`;
