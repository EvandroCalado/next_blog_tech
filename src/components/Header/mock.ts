import { HeaderProps } from '.';

export default {
  blogName: 'Evandro Calado',
  blogDescription:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Unde harum optio, eius quia quas molestias ipsum modi sed ullam exercitationem.',
  logo: '/assets/images/evandro.jpeg',
  showText: true,
} as HeaderProps;
