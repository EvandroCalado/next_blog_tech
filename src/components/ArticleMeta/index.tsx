import { Author } from '../shared-types/author';
import { Category } from '../shared-types/categories';
import { formatDate } from '../utils/format-date';
import * as Styled from './styles';

export type ArticleMetaProps = {
  createdAt: string;
  author?: Author;
  categories?: Category[];
};
const ArticleMeta = ({
  createdAt,
  author = undefined,
  categories = [],
}: ArticleMetaProps) => {
  return (
    <Styled.Wrapper>
      <p>
        {typeof author !== 'undefined' && (
          <>
            <span>Por</span>
            <a href={`/author/${author.slug}`}>{author.displayName}</a>
            <span className="separator"> | </span>
          </>
        )}
        <time dateTime={createdAt}>{formatDate(createdAt)}</time>
        {categories.length > 0 && (
          <>
            <span className="separator"> | </span>
            <span className="categories">
              {categories.map((category, index) => {
                return (
                  <span key={`article-meta-cat${index}`}>
                    <a href={`/category/${category.attributes.slug}`}>
                      {category.attributes.displayName}
                    </a>
                  </span>
                );
              })}
            </span>
          </>
        )}
      </p>
    </Styled.Wrapper>
  );
};

export default ArticleMeta;
