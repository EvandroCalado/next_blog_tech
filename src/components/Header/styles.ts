import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacings.xhuge};
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.dark};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${Title} {
      margin: ${theme.spacings.small} 0;
      text-align: left;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
