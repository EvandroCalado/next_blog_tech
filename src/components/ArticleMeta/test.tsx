import ArticleMeta from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

describe('<ArticleMeta />', () => {
  it('should render author and category links', () => {
    renderTheme(<ArticleMeta {...mock} />);

    expect(
      screen.getByRole('link', { name: 'Evandro Calado' }),
    ).toHaveAttribute('href', '/author/evandro-calado');

    expect(screen.getByRole('link', { name: 'React' })).toHaveAttribute(
      'href',
      '/category/react',
    );

    expect(screen.getByRole('link', { name: 'Next' })).toHaveAttribute(
      'href',
      '/category/next',
    );
  });

  it('should format date', () => {
    renderTheme(<ArticleMeta {...mock} />);

    expect(screen.getByText('27 de mar. de 2023')).toHaveAttribute(
      'datetime',
      mock.createdAt,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<ArticleMeta {...mock} />);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot without author and categories', () => {
    const { container } = renderTheme(
      <ArticleMeta {...mock} author={undefined} categories={undefined} />,
    );

    expect(container).toMatchSnapshot();
  });
});
