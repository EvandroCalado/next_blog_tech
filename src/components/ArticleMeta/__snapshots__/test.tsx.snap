// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleMeta /> should match snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #aaa;
  font-style: italic;
}

.c0 .categories span:not(:last-child)::after {
  content: ', ';
}

.c0 span:first-child {
  margin-right: 0.5rem;
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <div
    class="c0"
  >
    <p>
      <span>
        Por
      </span>
      <a
        href="/author/evandro-calado"
      >
        Evandro Calado
      </a>
      <span
        class="separator"
      >
         | 
      </span>
      <time
        datetime="2023-03-27T17:22:39.749Z"
      >
        27 de mar. de 2023
      </time>
      <span
        class="separator"
      >
         | 
      </span>
      <span
        class="categories"
      >
        <span>
          <a
            href="/category/react"
          >
            React
          </a>
        </span>
        <span>
          <a
            href="/category/next"
          >
            Next
          </a>
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[
  `<ArticleMeta /> should match snapshot without author and categories 1`
] = `
.c0 {
  font-size: 1.6rem;
  color: #aaa;
  font-style: italic;
}

.c0 .categories span:not(:last-child)::after {
  content: ', ';
}

.c0 span:first-child {
  margin-right: 0.5rem;
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <div
    class="c0"
  >
    <p>
      <time
        datetime="2023-03-27T17:22:39.749Z"
      >
        27 de mar. de 2023
      </time>
    </p>
  </div>
</div>
`;
